### PR TITLE
Flask-RESTplus GET APIs

### PR DESCRIPTION
**Pull request by shivamMg**
https://github.com/fossasia/open-event-orga-server/issues/355, https://github.com/fossasia/open-event-orga-server/issues/356 & https://github.com/fossasia/open-event-orga-server/issues/362

### CODE REVIEW

File: open_event/api/levels.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+})
+
+
+@api.route('/events/<int:event_id>/levels/<int:id>')
+@api.response(404, 'Level not found')
+@api.response(400, 'Object does not belong to event')
+class Level(Resource):
+    @api.doc('get_level')
+    @api.marshal_with(level)
+    def get(self, event_id, id):
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/levels.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.session import Level as LevelModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('levels', description='levels', path='/')
+
+level = api.model('level', {
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Invalid constant name "level"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l9)

File: open_event/api/levels.py (hunk 1 above)
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Redefining built-in 'id'](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/levels.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@api.response(400, 'Object does not belong to event')
+class Level(Resource):
+    @api.doc('get_level')
+    @api.marshal_with(level)
+    def get(self, event_id, id):
+        """Fetch a level given its id"""
+        return get_object_in_event(LevelModel, id, event_id)
+
+
+@api.route('/events/<int:event_id>/levels')
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l27)

File: open_event/api/levels.py (hunk 1 above)
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Invalid argument name "id"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/levels.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.session import Level as LevelModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('levels', description='levels', path='/')
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l7)

File: open_event/api/levels.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+level = api.model('level', {
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+    'label_en': fields.String,
+})
+
+
+@api.route('/events/<int:event_id>/levels/<int:id>')
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l16)

File: open_event/api/levels.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        """Fetch a level given its id"""
+        return get_object_in_event(LevelModel, id, event_id)
+
+
+@api.route('/events/<int:event_id>/levels')
+@api.param('event_id')
+class LevelList(Resource):
+    @api.doc('list_levels')
+    @api.marshal_list_with(level)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859642/issues/source?bid=3233864&fileBranchId=3314204#l32)

File: open_event/api/tracks.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+})
+
+
+@api.route('/events/<int:event_id>/tracks/<int:id>')
+@api.response(404, 'Track not found')
+@api.response(400, 'Object does not belong to event')
+class Track(Resource):
+    @api.doc('get_track')
+    @api.marshal_with(track)
+    def get(self, event_id, id):
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859647/issues/source?bid=3233864&fileBranchId=3314204#l29)

File: open_event/api/tracks.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        """Fetch a track given its id"""
+        return get_object_in_event(TrackModel, id, event_id)
+
+
+@api.route('/events/<int:event_id>/tracks')
+class TrackList(Resource):
+    @api.doc('list_tracks')
+    @api.marshal_list_with(track)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620859647/issues/source?bid=3233864&fileBranchId=3314204#l38)

File: open_event/api/formats.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.session import Format as FormatModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('formats', description='formats', path='/')
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l7)

File: open_event/api/formats.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+format_ = api.model('Format', {
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+    'label_en': fields.String,
+})
+
+
+@api.route('/events/<int:event_id>/formats/<int:id>')
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l16)

File: open_event/api/formats.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+@api.response(404, 'Format not found')
+@api.response(400, 'Object does not belong to event')
+class Format(Resource):
+    @api.doc('get_format')
+    @api.marshal_with(format_)
+    def get(self, event_id, id):
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/formats.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        """Fetch a format given its id"""
+        return get_object_in_event(FormatModel, id, event_id)
+
+
+@api.route('/events/<int:event_id>/formats')
+@api.param('event_id')
+class FormatList(Resource):
+    @api.doc('list_formats')
+    @api.marshal_list_with(format_)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l32)

File: open_event/api/formats.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.session import Format as FormatModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('formats', description='formats', path='/')
+
+format_ = api.model('Format', {
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Invalid constant name "format_"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l9)

File: open_event/api/formats.py (hunk 11 above)
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Invalid argument name "id"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/formats.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+@api.response(400, 'Object does not belong to event')
+class Format(Resource):
+    @api.doc('get_format')
+    @api.marshal_with(format_)
+    def get(self, event_id, id):
+        """Fetch a format given its id"""
+        return get_object_in_event(FormatModel, id, event_id)
+
+
+@api.route('/events/<int:event_id>/formats')
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l27)

File: open_event/api/formats.py (hunk 11 above)
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Redefining built-in 'id'](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934160/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/levels.py (hunk 1 above)
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934163/issues/source?bid=3233864&fileBranchId=3314204#l22)

File: open_event/api/levels.py (hunk 2 above)
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Invalid constant name "level"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620934163/issues/source?bid=3233864&fileBranchId=3314204#l9)

File: open_event/api/events.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+})
+
+
+@api.route('/<int:event_id>')
+@api.param('event_id')
+@api.response(404, 'Event not found')
+class Event(Resource):
+    @api.doc('get_event')
+    @api.marshal_with(_event)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l30)

File: open_event/api/events.py [hunk 16]
```diff
@@ -0,0 +1,41 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404
+
+api = Namespace('events', description='Events')
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l6)

File: open_event/api/events.py [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+    'end_time': fields.DateTime,
+    'latitude': fields.Float,
+    'longitude': fields.Float,
+    'slogan': fields.String,
+    'url': fields.String,
+    'location_name': fields.String,
+})
+
+
+@api.route('/<int:event_id>')
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l24)

File: open_event/api/events.py [hunk 18]
```diff
@@ -0,0 +1,41 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404
+
+api = Namespace('events', description='Events')
+
+_event = api.model('Event', {
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Invalid constant name "_event"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l8)

File: open_event/api/events.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    def get(self, event_id):
+        """Fetch an event given its id"""
+        return get_object_or_404(EventModel, event_id)
+
+
+@api.route('/')
+class EventList(Resource):
+    @api.doc('list_events')
+    @api.marshal_list_with(_event)
+    def get(self):
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l39)

File: open_event/api/events.py [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+@api.response(404, 'Event not found')
+class Event(Resource):
+    @api.doc('get_event')
+    @api.marshal_with(_event)
+    def get(self, event_id):
+        """Fetch an event given its id"""
+        return get_object_or_404(EventModel, event_id)
+
+
+@api.route('/')
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995756/issues/source?bid=3233864&fileBranchId=3314204#l35)

File: open_event/api/sessions.py [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+})
+
+
+@api.route('/events/<int:event_id>/sessions/<int:session_id>')
+@api.response(404, 'Session not found')
+@api.response(400, 'Object does not belong to event')
+class Session(Resource):
+    @api.doc('get_session')
+    @api.marshal_with(_session)
+    def get(self, event_id, session_id):
```
Comment (juslee) [inline open_event/api/sessions.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995762/issues/source?bid=3233864&fileBranchId=3314204#l57)

File: open_event/api/sessions.py [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+    'end_time': fields.DateTime,
+    'track': fields.Nested(_track),
+    'speakers': fields.List(fields.Nested(_speaker)),
+    'level': fields.Nested(_level),
+    'language': fields.Nested(_language),
+    'microlocation': fields.Nested(_microlocation),
+})
+
+
+@api.route('/events/<int:event_id>/sessions/<int:session_id>')
```
Comment (juslee) [inline open_event/api/sessions.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995762/issues/source?bid=3233864&fileBranchId=3314204#l51)

File: open_event/api/sessions.py [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+    'label_en': fields.String,
+})
+
+_language = api.model('Language', {
+    'id': fields.Integer(required=True),
+    'label_en': fields.String,
+    'label_de': fields.String,
+})
+
+_microlocation = api.model('Microlocation', {
```
Comment (juslee) [inline open_event/api/sessions.py]: Codacy Issue found: [Invalid constant name "_microlocation"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995762/issues/source?bid=3233864&fileBranchId=3314204#l30)

File: open_event/api/sessions.py [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+})
+
+_speaker = api.model('Speaker', {
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+})
+
+_level = api.model('Level', {
```
Comment (juslee) [inline open_event/api/sessions.py]: Codacy Issue found: [Invalid constant name "_level"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2620995762/issues/source?bid=3233864&fileBranchId=3314204#l19)

File: open_event/api/__init__.py [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+from .speakers import api as speaker_api
+from .sponsors import api as sponsor_api
+from .microlocations import api as microlocation_api
+from .levels import api as level_api
+from .formats import api as format_api
+from .languages import api as language_api
+
+api_v2 = Blueprint('api', __name__, url_prefix='/api/v2')
+
+api = Api(api_v2, version='2.0', title='Organizer Server APIs',
```
Comment (juslee) [inline open_event/api/__init__.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092603/issues/source?bid=3233864&fileBranchId=3314204#l16)

File: open_event/api/__init__.py [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+from .sessions import api as session_api
+from .tracks import api as track_api
+from .speakers import api as speaker_api
+from .sponsors import api as sponsor_api
+from .microlocations import api as microlocation_api
+from .levels import api as level_api
+from .formats import api as format_api
+from .languages import api as language_api
+
+api_v2 = Blueprint('api', __name__, url_prefix='/api/v2')
```
Comment (juslee) [inline open_event/api/__init__.py]: Codacy Issue found: [Invalid constant name "api_v2"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092603/issues/source?bid=3233864&fileBranchId=3314204#l14)

File: open_event/api/microlocations.py [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+    'latitude': fields.Float,
+    'longitude': fields.Float,
+    'floor': fields.Integer,
+    'room': fields.String,
+})
+
+
+@api.route('/events/<int:event_id>/microlocations/<int:microlocation_id>')
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092609/issues/source?bid=3233864&fileBranchId=3314204#l19)

File: open_event/api/microlocations.py [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+@api.response(404, 'Microlocation not found')
+@api.response(400, 'Object does not belong to event')
+class Microlocation(Resource):
+    @api.doc('get_microlocation')
+    @api.marshal_with(MICROLOCATION)
+    def get(self, event_id, microlocation_id):
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092609/issues/source?bid=3233864&fileBranchId=3314204#l25)

File: open_event/api/microlocations.py [hunk 29]
```diff
@@ -0,0 +1,41 @@
...
+        """Fetch a microlocation given its id"""
+        return get_object_in_event(MicrolocationModel, microlocation_id,
+                                   event_id)
+
+
+@api.route('/events/<int:event_id>/microlocations')
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092609/issues/source?bid=3233864&fileBranchId=3314204#l31)

File: open_event/api/microlocations.py [hunk 30]
```diff
@@ -0,0 +1,41 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.microlocation import Microlocation as MicrolocationModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('microlocations', description='microlocations', path='/')
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092609/issues/source?bid=3233864&fileBranchId=3314204#l7)

File: open_event/api/microlocations.py [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+        return get_object_in_event(MicrolocationModel, microlocation_id,
+                                   event_id)
+
+
+@api.route('/events/<int:event_id>/microlocations')
+@api.param('event_id')
+class MicrolocationList(Resource):
+    @api.doc('list_microlocations')
+    @api.marshal_list_with(MICROLOCATION)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092609/issues/source?bid=3233864&fileBranchId=3314204#l36)

File: open_event/api/tracks.py [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+    def get(self, event_id, track_id):
+        """Fetch a track given its id"""
+        return get_object_in_event(TrackModel, track_id, event_id)
+
+
+@api.route('/events/<int:event_id>/tracks')
+class TrackList(Resource):
+    @api.doc('list_tracks')
+    @api.marshal_list_with(TRACK)
+    def get(self, event_id):
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092613/issues/source?bid=3233864&fileBranchId=3314204#l38)

File: open_event/api/tracks.py [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+TRACK = api.model('Track', {
+    'id': fields.Integer(required=True),
+    'name': fields.String,
+    'description': fields.String,
+    'track_image_url': fields.String,
+    'sessions': fields.List(fields.Nested(SESSION)),
+})
+
+
+@api.route('/events/<int:event_id>/tracks/<int:track_id>')
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092613/issues/source?bid=3233864&fileBranchId=3314204#l23)

File: open_event/api/tracks.py [hunk 34]
```diff
@@ -0,0 +1,43 @@
+from flask.ext.restplus import Resource, Namespace, fields
+
+from open_event.models.track import Track as TrackModel
+from open_event.models.event import Event as EventModel
+from .helpers import get_object_list, get_object_or_404, get_object_in_event
+
+api = Namespace('tracks', description='Tracks', path='/')
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Invalid constant name "api"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2621092613/issues/source?bid=3233864&fileBranchId=3314204#l7)

**Comment by shivamMg** [comment]
@juslee Please review.

The "Too few public methods (1/2)" errors are occurring because only the `get` method has been defined right now. After we add `post`, `put` and `delete` methods they will not occur. 

"Method could be a function": restplus requires them to be a method [ex](https://github.com/noirbizarre/flask-restplus/blob/master/examples/todo_blueprint.py#L42).

I've solved all the other issues.

**Comment by juslee** [comment]
Looks good. Resolve them in the next PR. I'll merge. @shivamMg Do you have the swagger api docs hosted somewhere?

Good job.

**Comment by shivamMg** [comment]
@juslee The swagger docs can be viewed in the local installation: http://127.0.0.1:8001/api/v2/

Corresponding swagger config: http://127.0.0.1:8001/api/v2/swagger.json

**Pull request merged**